### PR TITLE
Add Swift code style in English

### PR DESCRIPTION
**Pull request by giginet**
refs #40 

I translated #40 to English.

Could you point to my poor English skill? 😕

### CODE REVIEW

File: swift.en.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Do not use `NSException`.
+- **[SHOULD]** Do not use `try!`. Use `do` ~ `catch` instead.
+
+## Naming
+
+- **[MUST]** Follow [Swift API Design Guideline](https://swift.org/documentation/api-design-guidelines/)
+- **[MUST]** Use `_` for unused returning values or arguments.
+- **[MUST]** Do not use vendor prefix.
+- **[MUST]** Follow same naming rule for constants as variables.
+- **[MUST]** To define constants, follow naming rule of variables.
```
Comment (giginet) [inline swift.en.md]: duplicated

File: swift.en.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+## Introduction
+
+This document defines the conventions for writing Swift code at Cookpad Inc.
+
+This guide directed for Swift 3.0.2.
+
+## Objective
+
+This guides aim to accomplish these objectives.
```
Comment (pauchan) [inline swift.en.md]: I would suggest s/these/following
And also s/./: at the end of the line

File: swift.en.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+1. Clarify purposes 
+1. Improve readability
+1. Keep consistency
+1. Reduce side-effects
+
+## Code style
+
+- **[MUST]** Use four spaces for 1-level of indent.  Do not use the horizontal tab character.
+- **[MUST]** Add linebreak to end of files
```
Comment (pauchan) [inline swift.en.md]: s/to/at the

Comment (pauchan) [inline swift.en.md]: Also I think it would be good to be consistent with period at the end of the sentences. This line and the next line are missing periods.

File: swift.en.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+{
+}
+
+// Good
+func update() {
+}
+```
+
+- **[MUST]** Use trailing comma on elements of collections
+    - Because of reducing file diffs
```
Comment (pauchan) [inline swift.en.md]: This will reduce file diffs in the future

File: swift.en.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+// Good
+let ingredients = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+```
+
+- **[MUST]** Do not use `()` wraps condition statements
```
Comment (pauchan) [inline swift.en.md]: Do not use external `()` to wrap conditional statements

File: swift.en.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+// Bad
+if (a == 0) {
+}
+
+// Good
+if a == 0 {
+}
+```
+
+- **[MUST]** Do not use `;` on end of lines
```
Comment (pauchan) [inline swift.en.md]: s/on/at the

File: swift.en.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+
+## Class, Struct
```
Comment (pauchan) [inline swift.en.md]: maybe it would be a good idea to use plural here?
So `Classes, Structs`

File: swift.en.md [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+// Good
+func update() {
+}
+```
+
+## Closure
+
+- **[MUST]** Use trailing closure if the closure has just one argument.
```
Comment (pauchan) [inline swift.en.md]: maybe `exactly one` would be better here?

File: swift.en.md [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+// Good
+if a == 0 {
+}
+```
+
+- **[MUST]** Do not use `;` on end of lines
+
+## Class, Struct
+
+- **[MUST]** Do not inherit `NSObject` except if needed to refer from Objective-C.
```
Comment (pauchan) [inline swift.en.md]: how about
Unless reference from Objective-C is needed, do not inherit from `NSObject`
?

File: swift.en.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+- **[SHOULD]** Do not use `mutating` attribute for structs.
+    - Because structs should be immutable.
```
Comment (pauchan) [inline swift.en.md]: It seems that it make sense to split this part in two sentences but for English maybe we can combine it into one like:
`Since structs should be immutable, do not use `mutating` attribute for structs.`

File: swift.en.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+
+## Method
+
+- **[MUST]** Use closest access level as possible.
```
Comment (pauchan) [inline swift.en.md]: 狭い in this context is hard to translate directly into English but maybe instead of `closest` we can use `most local`?
CC @halogenandtoast

File: swift.en.md [hunk 12]
```diff
@@ -0,0 +1,340 @@
+# Swift coding style
+
+## Introduction
+
+This document defines the conventions for writing Swift code at Cookpad Inc.
```
Comment (pauchan) [inline swift.en.md]: maybe we can drop Inc here?
sounds super formal

File: swift.en.md [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+
+This guide directed for Swift 3.0.2.
```
Comment (halogenandtoast) [inline swift.en.md]: This guide is for Swift 3.0.2

File: swift.en.md (hunk 2 above)
Comment (halogenandtoast) [inline swift.en.md]: This guide aims to accomplish the following objectives:

File: swift.en.md [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+
+## Objective
+
+This guides aim to accomplish these objectives.
+
+1. Clarify purposes
```
Comment (halogenandtoast) [inline swift.en.md]: 1. Clarify purpose

File: swift.en.md [hunk 15]
```diff
@@ -0,0 +1,340 @@
...
+## Class, Struct
+
+- **[MUST]** Do not inherit `NSObject` except if needed to refer from Objective-C.
+- **[SHOULD]** Do not use `mutating` attribute for structs.
+    - Because structs should be immutable.
+
+## Method
+
+- **[MUST]** Use closest access level as possible.
+- **[MUST]** Do not use `Void` as returning type.
```
Comment (pauchan) [inline swift.en.md]: I think it's:
s/returning type/return type

https://en.wikipedia.org/wiki/Return_type

File: swift.en.md [hunk 16]
```diff
@@ -0,0 +1,340 @@
...
+This guides aim to accomplish these objectives.
+
+1. Clarify purposes 
+1. Improve readability
+1. Keep consistency
+1. Reduce side-effects
+
+## Code style
+
+- **[MUST]** Use four spaces for 1-level of indent.  Do not use the horizontal tab character.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use four spaces for 1-level of indentation.  Do not use the horizontal tab character.

OR EVEN BETTER

**[MUST]** Use two spaces for 1-level of indentation.  Do not use the horizontal tab character.

File: swift.en.md [hunk 17]
```diff
@@ -0,0 +1,340 @@
...
+// Bad
+client.search(keyword: "Sushi", completion: { recipes in
+})
+
+// Good
+client.search(keyword: "Sushi") { recipes in
+}
+```
+
+- **[MUST]** Do not use `()` when calling the closure has just one argument.
```
Comment (pauchan) [inline swift.en.md]: Im not sure but maybe we can drop `the` here.
Also
`s/closure has/closure that has`

File: swift.en.md (hunk 4 above)
Comment (halogenandtoast) [inline swift.en.md]: I still don't understand why people actually care about this. It's so extremely trite.

File: swift.en.md [hunk 18]
```diff
@@ -0,0 +1,340 @@
...
+```swift
+// Bad
+let titles = recipes.map() { $0.title }
+
+// Good
+let titles = recipes.map { $0.title }
+```
+
+- **[MUST]** Use shorthand argument if closure defined in one line.
+    - In contrast, Do not use it if closure defined in multiple lines.
```
Comment (pauchan) [inline swift.en.md]: s/Do/do
s/closure defined/closure is defined

File: swift.en.md [hunk 19]
```diff
@@ -0,0 +1,340 @@
...
+
+// Good
+client.search(keyword: "Sushi") { results, error in
+    if error != nil {
+        recipes = results
+    }
+}
+```
+
+- **[MUST]** Do not use `@escaping` without needed
```
Comment (pauchan) [inline swift.en.md]: s/without needed/if not needed

File: swift.en.md (hunk 9 above)
Comment (halogenandtoast) [inline swift.en.md]: Unless a reference from Objective-C is needed, do not inherit from NSObject

File: swift.en.md (hunk 10 above)
Comment (halogenandtoast) [inline swift.en.md]: Because structs should be immutable, do not use the `mutating` attribute for structs.

File: swift.en.md [hunk 20]
```diff
@@ -0,0 +1,340 @@
...
+
+```swift
+// Bad
+let titles = recipes.map() { $0.title }
+
+// Good
+let titles = recipes.map { $0.title }
+```
+
+- **[MUST]** Use shorthand argument if closure defined in one line.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use shorthand argument if closure is defined on a single line.

File: swift.en.md (hunk 18 above)
Comment (halogenandtoast) [inline swift.en.md]: - In contrast, do not use it if the closure is defined on multiple lines.

File: swift.en.md (hunk 17 above)
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use `()` when the calling closure has just one argument.

File: swift.en.md (hunk 11 above)
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use the closest access level possible.

Comment (halogenandtoast) [inline swift.en.md]: Could use an example.

File: swift.en.md (hunk 19 above)
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use `@escaping` unless necessary

File: swift.en.md [hunk 21]
```diff
@@ -0,0 +1,340 @@
...
+client.search(keyword: "Sushi") { results, error in
+    if error != nil {
+        recipes = results
+    }
+}
+```
+
+- **[MUST]** Do not use `@escaping` without needed
+- **[MUST]** Use `weak` instead of `unowned` as capture controls.
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
```
Comment (halogenandtoast) [inline swift.en.md]: - While `unowned` can sometimes improve performance,  readability will be lost.

File: swift.en.md [hunk 22]
```diff
@@ -0,0 +1,340 @@
...
+
+## Property, Variable
+
+- **[MUST]** Use `let` to make variables immutable as possible.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use `let` to make variables immutable when possible.

File: swift.en.md [hunk 23]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Drop type definition as possible. However, in the case of avoiding type inference for improvement compilation time, you can define clearly.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Drop type definitions when possible. However, when trying to avoid type inferencing for faster compilation time, you may choose to use them.

File: swift.en.md [hunk 24]
```diff
@@ -0,0 +1,340 @@
...
+
+### Property
+
+- **[MUST]** Use closest access control as possible.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use closest access control possible.

File: swift.en.md [hunk 25]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Use `private(set)` for the property want to write from limited scopes.
```
Comment (halogenandtoast) [inline swift.en.md]: Not sure what this is trying to say.

File: swift.en.md [hunk 26]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Use `weak` as property attributes as possible.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use `weak` for property attributes when possible.

File: swift.en.md [hunk 27]
```diff
@@ -0,0 +1,340 @@
...
+    - You should use `weak` to avoid circular references (eg. `@IBOutlet`, delegate)
+- **[MUST]** Do not use `unowned` as property attribute.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use the `unowned` property attribute.

File: swift.en.md [hunk 28]
```diff
@@ -0,0 +1,340 @@
...
+    - In some cases, `unowned` improves performance. In other hand readability will be lost.
```
Comment (halogenandtoast) [inline swift.en.md]: - In some cases, `unowned` improves performance. On the other hand, readability will be lost.

File: swift.en.md [hunk 29]
```diff
@@ -0,0 +1,340 @@
...
+// Bad
+var delegate: FooDelegate
+@IBOutlet var textLabel: UILabel!
+
+// Good
+weak var delegate: FooDelegate
+@IBOutlet weak var textLabel: UILabel!
+```
+
+- **[MUST]** Do not use `get` block as possible on computed properties.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use a `get` block when possible on computed properties.

File: swift.en.md [hunk 30]
```diff
@@ -0,0 +1,340 @@
...
+```
+
+- **[SHOULD]** Use `typealias` to represent complex type definitions.
+
+```swift
+// Good
+typealias RecipeClientCompletionBlock = (Result<[Recipe], APIError>) -> Void
+```
+
+- **[SHOULD]** Use classes without `NS` prefixes as possible.
```
Comment (halogenandtoast) [inline swift.en.md]: **[SHOULD]** Use classes without `NS` prefixes when possible.

File: swift.en.md [hunk 31]
```diff
@@ -0,0 +1,340 @@
...
+let url = URL(string: "https://cookpad.com/")
+```
+
+
+- **[SHOULD]** Do not use `UInt`
+    - [Apple announced to avoid using it.](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID320)
+
+## Enum
+
+- **[MUST]** Use lowerCamelCase as value naming.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use lowerCamelCase for value names.

File: swift.en.md [hunk 32]
```diff
@@ -0,0 +1,340 @@
...
+user.status = UserStatus.guest
+
+// Good
+user.status = .guest
+```
+
+## Optional
+
+- **[MUST]** Do not use `Optional<T>`. Use syntax sugar instead.
+- **[MUST]** Use non-Optional value as possible.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use non-Optional value when possible.

File: swift.en.md [hunk 33]
```diff
@@ -0,0 +1,340 @@
...
+
+## Exception
+
+- **[MUST]** Do not use `NSException`.
+- **[SHOULD]** Do not use `try!`. Use `do` ~ `catch` instead.
+
+## Naming
+
+- **[MUST]** Follow [Swift API Design Guideline](https://swift.org/documentation/api-design-guidelines/)
+- **[MUST]** Use `_` for unused returning values or arguments.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Use `_` for unused return values or arguments.

File: swift.en.md [hunk 34]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Do not use vendor prefix.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use vendor prefixes.

File: swift.en.md [hunk 35]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Follow same naming rule for constants as variables.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Follow the same naming rules for constants as variables.

File: swift.en.md [hunk 36]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** To define constants, follow naming rule of variables.
+    - Do not use UpperCamelCase, SNAKE_CASE or `k` prefix as constants names.
```
Comment (halogenandtoast) [inline swift.en.md]: - Do not use UpperCamelCase, SNAKE_CASE or `k` prefix when naming constants.

File: swift.en.md [hunk 37]
```diff
@@ -0,0 +1,340 @@
...
+- **[MUST]** Do not use non-Ascii characters as variable names.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use non-Ascii characters in variable names.

File: swift.en.md [hunk 38]
```diff
@@ -0,0 +1,340 @@
...
+- **[SHOULD]** Use nest structure to reperesent hierarchy.
```
Comment (halogenandtoast) [inline swift.en.md]: **[SHOULD]** Use nested structures to represent hierarchy.

File: swift.en.md (hunk 4 above)
Comment (kittinunf) [inline swift.en.md]: Why is this a must? I don't see much of a gain by having unnecessary `,` hanging around at the end of the line.

File: swift.en.md [hunk 39]
```diff
@@ -0,0 +1,340 @@
...
+        case none
+    }
+}
+
+```
+
+
+## Syntax
+
+- **[MUST]** Do not use `self` as possible. However, you can use it like following situations.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use `self` when possible. However, you can use it in the following situations:

File: swift.en.md [hunk 40]
```diff
@@ -0,0 +1,340 @@
...
+  - Assign value from variable has same names
```
Comment (halogenandtoast) [inline swift.en.md]: - Assign value and variable have the same names

File: swift.en.md [hunk 41]
```diff
@@ -0,0 +1,340 @@
...
+  - Call method in closures
+- **[MUST]** Do not use syntaxes which will be deprecated in the future.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use any syntax which will be deprecated in the future.

File: swift.en.md [hunk 42]
```diff
@@ -0,0 +1,340 @@
...
+
+## Cocoa
+
+= **[MUST]** Do not use factory method. Use constructors instead.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use factory methods. Use constructors instead.

File: swift.en.md [hunk 43]
```diff
@@ -0,0 +1,340 @@
...
+
+```swift
+// Bad
+let rect = CGRectMake(10.0, 20.0, 30.0, 40.0)
+
+// Good
+let rect = CGRect(x: 10.0, y: 20.0, width: 30.0, height: 40.0)
+```
+
+- **[MUST]** Do not use class method to create new instances. Use constructors.
```
Comment (halogenandtoast) [inline swift.en.md]: **[MUST]** Do not use class methods to create new instances. Use constructors.

File: swift.en.md (hunk 11 above)
Comment (kittinunf) [inline swift.en.md]: I would say, **Use the most restrictive access modifier if possible**.

File: swift.en.md (hunk 15 above)
Comment (kittinunf) [inline swift.en.md]: I think, if it is in the case of `typealias` like `var handler: () -> Void = { _ in }`. It should be exempted.

File: swift.en.md (hunk 8 above)
Comment (kittinunf) [inline swift.en.md]: Not sure about this. Does this suggest

``` Swift
// Bad
client.search(with: "abc") { response, result in

}
```

Because in this case, the trailing closure has 2 arguments.

Comment (pauchan) [inline swift.en.md]: Ah I guess the important part of the original sentence got lost in translation.
It should state that you must use trailing closure if the method signature has exactly one closure parameter.

File: swift.en.md (hunk 11 above)
Comment (pauchan) [inline swift.en.md]: I like it!

File: swift.en.md (hunk 16 above)
Comment (pauchan) [inline swift.en.md]: why ALL CAPS? :)

File: swift.en.md (hunk 25 above)
Comment (pauchan) [inline swift.en.md]: That you must use `private(set)` when the property is readwrite from inside the scope and readonly from outside

File: swift.en.md (hunk 28 above)
Comment (pauchan) [inline swift.en.md]: I don't like this `On the other hand, readability will be lost.`
I think it should be somehow combined into one sentence, like `In some cases, unowned improves performance but it also decreases readability.

File: swift.en.md [hunk 44]
```diff
@@ -0,0 +1,340 @@
...
+- **[SHOULD]** If setting a property has a side-effect, write it in `willSet` or `didSet`.
+
+### Variable
+
+- **[MUST]** Do not use global variables.
+
+
+## Type
+
+- **[MUST]** Do not use `Array<T>`, `Dictionary<K, V>`. Use syntax sugar instead.
```
Comment (Ecarrion) [inline swift.en.md]: Syntactic sugar maybe?

File: swift.en.md [hunk 45]
```diff
@@ -0,0 +1,340 @@
...
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+```
+
+- **[SHOULD]** Use `typealias` to represent complex type definitions.
+
+```swift
+// Good
+typealias RecipeClientCompletionBlock = (Result<[Recipe], APIError>) -> Void
```
Comment (Ecarrion) [inline swift.en.md]: Should it be `Closure` instead of `Block`?

File: swift.en.md [hunk 46]
```diff
@@ -0,0 +1,340 @@
...
+```swift
+// Bad
+let seasonings: Array<String> = ["sugar", "salt", "vinegar", "soy sauce", "miso"]
+let ingredients: Dictionary<String, String> = [
+    "bacon": "100g",
+    "lettuce": "1/4",
+    "tomato": "half",
+]
+
+// Good let seasonings: [String] = ["sugar", "salt", "vinegar", "soy sauce", "miso"] let ingredients: [String: String] = [ "bacon": "100g",
```
Comment (kittinunf) [inline swift.en.md]: Are we missing one (and very important line break)? 
Also the fact that we need to specify the type like `: [String]` is kinda breaking the rule above.

File: swift.en.md [hunk 47]
```diff
@@ -0,0 +1,340 @@
...
+- **[SHOULD]** Use classes without `NS` prefixes as possible.
+
+```swift
+// Bad
+let url = NSURL(string: "https://cookpad.com/")
+
+// Good
+let url = URL(string: "https://cookpad.com/")
+```
+
```
Comment (kittinunf) [inline swift.en.md]: Unnecessary line break here.

File: swift.en.md [hunk 48]
```diff
@@ -0,0 +1,340 @@
...
+// Good
+user.status = .guest
+```
+
+## Optional
+
+- **[MUST]** Do not use `Optional<T>`. Use syntax sugar instead.
+- **[MUST]** Use non-Optional value as possible.
+- **[MUST]** Do not use `guard` without optional-binding.
```
Comment (kittinunf) [inline swift.en.md]: Not fully understand this line. I don't think code with `guard` would compile with non-optional value.

File: swift.en.md (hunk 10 above)
Comment (kittinunf) [inline swift.en.md]: Sometimes, having mutating attributes or function is kinda useful.

``` Swift
struct Rectangle {
    var topLeft: Point
    var size: Size
}

extension Rectangle {
    mutating translate(with offset: Point) {
        origin += offset
    }
    
    func translated(with offset: Point) -> Rectangle {
        var _self = self
        _self.translate(offset)
        return _self
    }
}

var myRect = Rectangle(topLeft: Point(0, 0), size: Size(100, 200))
myRect.translate(with: Point(10, 10))

let anotherRect = myRect.tanslated(with: Point(20, 20))
```

This 2 scenarios are all well adopted by Apple such as `Array` that has both `sort` and `sorted`.

File: swift.en.md (hunk 43 above)
Comment (kittinunf) [inline swift.en.md]: "for instantiation" might sound a bit smoother.

File: swift.en.md (hunk 28 above)
Comment (kittinunf) [inline swift.en.md]: And also in the sense that we know exactly that this reference is gonna outlive you. Use `unowned` is a perfect keyword for that.

File: swift.en.md [hunk 49]
```diff
@@ -0,0 +1,340 @@
...
+
+```swift
+// Bad
+guard isLoggedIn else {
```
Comment (Ecarrion) [inline swift.en.md]: Can I ask the reason behind this? 😇

Comment (pauchan) [inline swift.en.md]: Don't you agree its a bad practice @Ecarrion ?
I think `notLoggedIn` is a expected path of the logic and as such, should not be guarded against

File: swift.en.md (hunk 12 above)
Comment (giginet) [inline swift.en.md]: Oh, I copied it from Ruby version 😥 
https://github.com/cookpad/styleguide/blob/master/ruby.en.md#introduction

I'll get rid of Inc.

File: swift.en.md (hunk 8 above)
Comment (giginet) [inline swift.en.md]: It was a wrong translation.

I meant use trailing closure when methods have exactly one closure as arguments.
It doesn't mean closure have exactly one argument.

```
- **[MUST]** Use trailing closure if the method has exactly one closure.
```

File: swift.en.md (hunk 46 above)
Comment (giginet) [inline swift.en.md]: oops 😢

File: swift.en.md (hunk 45 above)
Comment (giginet) [inline swift.en.md]: In JP team, we often name `Block` to represent closure.
Is it not common?

**Comment by giginet** [comment]
I fixed some grammatical issues.
Thank you for pointing out.

However, there are still some notable issues.
Let's discuss them.

File: swift.en.md (hunk 4 above)
Comment (giginet) [inline swift.en.md]: @halogenandtoast @kittinunf 

> Why is this a must? 

We think this rule helps to reduce unnecessary diff lines when PR and make it easy for reviewers.

There is the linter on SwiftLint.
We enabled this, people don't care about this anymore.

https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/TrailingCommaRule.swift

How do you think about?

File: swift.en.md (hunk 15 above)
Comment (giginet) [inline swift.en.md]: This section mentions about methods.
So I think these will be exempted in case of closures.

**Comment by pauchan** [comment]
Good job @giginet !
Thanks so much for doing this!
LGTM 🚀

**Comment by giginet** [comment]
@cookpad/global-ios Thank you for your time!
If you have any issues, feel free contact me.
